src/mtcp/mtcp_restart.c

```diff
@@ -861,6 +862,20 @@ static int read_one_memory_area(int fd)
     }
   }
 
+#ifdef FAST_RST_VIA_MMAP
+    /* CASE MAP_ANONYMOUS with FAST_RST enabled
+     * We only want to do this in the MAP_ANONYMOUS case, since we don't want
+     *   any writes to RAM to be reflected back into the underlying file.
+     * Note that in order to map from a file (ckpt image), we must turn off
+     *   anonymous (~MAP_ANONYMOUS).  It's okay, since the fd
+     *   should have been opened with read permission, only.
+     */
+    else if (area.flags & MAP_ANONYMOUS) {
+      mmapfile (fd, area.addr, area.size, area.prot | PROT_WRITE,
```

Comment:
@rohgarg, In fast restart, we want to mmap from the checkpoint image file to RAM (virtual memory).  When we'e done, we want the RAM to have read-write permission if it had read-write permission before checkpointing.  So in this case, we will never turn off the write permission for RAM.

On the other hand, we do not want the writes to RAM to be reflected all the way back into modifying the checkpoint image file itself.  So, ideally, we would choose MAP_ANONYMOUS mode.  But this is not possible, since that mode would not allow us to read from the checkpoint image file.  So, we must _not_ used MAP_ANONYMOUS.  So, to ensure that writes to RAM are not reflected back into modifying the checkpoint image file, we don't give write permission to the fd that points to the checkpoint image file.

So, in summary, the fd pointing to the checkpoint image file never has write permission.  On the other hand, the PROT_WRITE permission for the RAM is always turned on, since we always want to be able to write back to RAM and modify it.

We can add another statement in the comment if you like.  Suppose we wrote, "PROT_WRITE gives us permission to modify the RAM, while the lack of write permission for the fd ensures that we never write back to the checkpoint image file itself."

Would that comment make it clearer?